src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py

```diff
@@ -1080,3 +1080,37 @@ def test_aggregation_node_failed(self, pf):
         run_dict = run._to_dict()
         assert "error" in run_dict
         assert run_dict["error"] == exception
+
+    def test_get_details_against_partial_completed_run(self, pf: PFClient) -> None:
+        flow_mod2 = f"{FLOWS_DIR}/mod-n/two"
+        flow_mod3 = f"{FLOWS_DIR}/mod-n/three"
+        data_path = f"{DATAS_DIR}/numbers.jsonl"
+        # batch run against data
+        run1 = pf.run(
+            flow=flow_mod2,
+            data=data_path,
+            column_mapping={"number": "${data.value}"},
+        )
+        pf.runs.stream(run1)
+        details1 = pf.get_details(run1)
+        assert len(details1) == 20
+        assert len(details1.loc[details1["outputs.output"] != "(Failed)"]) == 10
+        # assert to ensure inputs and outputs are aligned
+        for _, row in details1.iterrows():
+            if str(row["outputs.output"]) != "(Failed)":
```

Comment:
We have a special str(Failed) now. might need to revisit this in future. 
How do user get the failed lines and error message.

Reply:
agreed, have opened a work item tracking this.